main.py

```diff
@@ -2,23 +2,90 @@
 #
 # Simple manager prototype for xqemu
 #
-from PyQt5.QtWidgets import QApplication, QDialog, QFileDialog, QMainWindow, QMessageBox
+from PyQt5.QtWidgets import QApplication, QDialog, QFileDialog, QMainWindow, QMessageBox, QTableWidgetItem, QAbstractItemView, QHeaderView
 from PyQt5.uic import loadUiType
 from PyQt5 import QtCore, QtGui
 from qmp import QEMUMonitorProtocol
+from glob import glob1
 import sys
 import os, os.path
 import json
 import subprocess
 import time
 import platform
+import csv
+import sys
 
 SETTINGS_FILE = './settings.json'
+LIBRARY_CACHE = './library.cache'
 
 # Load UI files
 settings_class, _ = loadUiType('settings.ui')
 mainwindow_class, _ = loadUiType('mainwindow.ui')
 
+class DiscImageFormat(object):
+	XISO = 'XISO'
+	ISO9660 = 'ISO9660'
+	Unknown = 'Unknown format'
+
+def read_from_offset(file, offset, length):
+	file.seek(offset)
+	return file.read(length).decode('unicode-escape')
+
+def getDiscFileFormat(filepath):
+	file = open(filepath, 'rb')
+
+	#XISO/XGD
+	if read_from_offset(file, 0x10000, 20) == 'MICROSOFT*XBOX*MEDIA': return DiscImageFormat.XISO
+
+	# ISO 9660
+	for offset in [0x9001, 0x8801, 0x8001]:
+		if read_from_offset(file, offset, 5) == 'CD001': return DiscImageFormat.ISO9660
+
+	return DiscImageFormat.Unknown
+
+class CacheManager(object):
```

Comment:
"CacheManager" feels rather ambiguous. What kind of Cache?
`checkExisting` clearly makes this into some kind of "FileIndexCacheManager" or something.

*(I'm also still trying to find out why this class is necessary)*